lib/jsonapi/utils.rb

```diff
@@ -245,7 +248,7 @@ def count_records(records, options)
 
     def setup_request
       @request ||=
-        JSONAPI::Request.new(
+        JSONAPI::RequestParser.new(
```

Comment:
Makes sense. I'll remove it from this PR, you can handle managing the separate branch for future bugfixes ye?